src/Adapters/Storage/Mongo/MongoStorageAdapter.js

```diff
@@ -113,26 +113,36 @@ export class MongoStorageAdapter implements StorageAdapter {
   _uri: string;
   _collectionPrefix: string;
   _mongoOptions: Object;
+  _onchange: any;
+  _stream: any;
   // Public
   connectionPromise: ?Promise<any>;
   database: any;
   client: MongoClient;
   _maxTimeMS: ?number;
   canSortOnJoinTables: boolean;
+  horizontalScaling: boolean;
 
   constructor({ uri = defaults.DefaultMongoURI, collectionPrefix = '', mongoOptions = {} }: any) {
     this._uri = uri;
     this._collectionPrefix = collectionPrefix;
     this._mongoOptions = mongoOptions;
     this._mongoOptions.useNewUrlParser = true;
     this._mongoOptions.useUnifiedTopology = true;
+    this._onchange = () => {};
 
     // MaxTimeMS is not a global MongoDB client option, it is applied per operation.
     this._maxTimeMS = mongoOptions.maxTimeMS;
     this.canSortOnJoinTables = true;
+    this.horizontalScaling = !!mongoOptions.horizontalScaling;
```

Comment:
It's in both places. The purpose of `horizontalScaling` is to enable the DB hooks. This is also useful for testing and prevents the introduction of flaky tests.

Reply:
For me it implies something different, depending on where it is:
- in the root it refers to parse server horizontal scaling, i.e. multiple server instances.
- In the DB options it refers to DB horizontal scaling, i.e. multiple DB clusters, i.e. sharding in MongoDB.

So I’m thinking:
- The term “horizontal scaling” In DB options should be reserved for options related to DB scaling or “sharding”, which is how horizontal scaling is understood in a MongoDB context.
- Does this option have to be in both places or would it be enough in the root?

Reply:
> Does this option have to be in both places or would it be enough in the root?

We just need a way to set a field to enable the db hooks.

```
const enableHooks = true;
const databaseAdapter = new PostgresStorageAdapter({
  uri: databaseURI,
  collectionPrefix: '',
  databaseOptions: {},
  enableHooks,
});
const api = new ParseServer({
  databaseAdapter
});
```
This could work

Reply:
What do you think about:

```
const databaseAdapter = new PostgresStorageAdapter({
  uri: databaseURI,
  collectionPrefix: '',
  databaseOptions: {},
  enableHooks: config.horizontalScaling,
});
```

Reply:
I updated with `enableSingleSchemaCache`

Reply:
Great, is this now under `database.enableSchemaHooks` or on the root level? If we need it only for the DB adapter, I think it could go into the `database` option. Sorry if I wasn't clear about this earlier.

Reply:
If we don't have it at the root level how will users set it at the database level?

Reply:
Through the database options:

```
const api = new ParseServer({
  databaseOptions: {
    enableSchemaHooks: true
  },
}
```